src/app/config/newTeam/newTeamReducer.js

```diff
@@ -0,0 +1,51 @@
+import { GET_USERS_REQUEST_SUCCESS, NEW_TEAM_ADD_USER, NEW_TEAM_REMOVE_USER_FROM, NEW_TEAM_RESET, NEW_TEAM_UNSAVED_CHANGES } from "../constants";
+
+export const initialState = {
+    usersInTeam: [],
+    usersNotInTeam: [],
+    allUsers: [],
+    unsavedChanges: false,
+};
+
+const newTeamReducer = (state = initialState, action) => {
+    switch (action.type) {
+        case GET_USERS_REQUEST_SUCCESS: {
+            return {
+                ...state,
+                allUsers: [...action.users],
+                usersNotInTeam: [...action.users],
+            };
+        }
+        case NEW_TEAM_UNSAVED_CHANGES: {
+            return {
+                ...state,
+                unsavedChanges: action.unsavedChanges,
+            };
+        }
+        case NEW_TEAM_REMOVE_USER_FROM: {
+            return {
+                ...state,
+                usersInTeam: state.usersInTeam.filter(filteredUser => filteredUser.email !== action.user.email),
+                usersNotInTeam: [...state.usersNotInTeam, action.user],
+            };
+        }
+        case NEW_TEAM_ADD_USER: {
+            return {
+                ...state,
+                usersNotInTeam: state.usersNotInTeam.filter(filteredUser => filteredUser.email !== action.user.email),
+                usersInTeam: [...state.usersInTeam, action.user],
+            };
+        }
+        case NEW_TEAM_RESET: {
+            return {
+                ...initialState,
```

Comment:
can you please do not mutate state in this way. I really do not understand what you are trying to achieve by doing this. you remove the whole state and there next line is to fetch it again? 
    const { dispatch, router, route, newTeam } = props;
    const [userConfirmedToLeave, setUserConfirmedToLeave] = useState(false);
    const [teamName, setTeamName] = useState("");
    useEffect(() => {
        dispatch(resetNewTeam());
        dispatch(getUsersRequest());
    }, []);